### ISSUE TITLE
Mysterywood saplings crashing the game.

### ISSUE THREAD

**Issue by Haemo-A** [simibubi/Above-and-Beyond#237]
Whenever a mysterywood sapling naturally grows, the game crashs and closes instantly. Additionally, if any bone meal is used on them, the sapling simply vanishes.

**Comment by oierbravo** [comment]
Same here.
Error log:
```
[09:05:21] [Server thread/FATAL] [ne.mi.co.ForgeMod/]: Preparing crash report with UUID 8454cd9a-5773-459e-9ec3-5005008ec022
[09:05:21] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
net.minecraft.crash.ReportedException: Exception ticking world
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:855) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:78) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292] {}
Caused by: java.lang.ClassCastException: net.minecraft.world.gen.feature.DecoratedFeatureConfig cannot be cast to net.minecraft.world.gen.feature.BaseTreeFeatureConfig
	at net.minecraft.block.trees.Tree.func_230339_a_(SourceFile:29) ~[?:?] {re:classloading}
	at net.minecraft.block.SaplingBlock.func_226942_a_(SaplingBlock.java:43) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.block.SaplingBlock.func_225542_b_(SaplingBlock.java:33) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.block.AbstractBlock$AbstractBlockState.func_227034_b_(AbstractBlock.java:624) ~[?:?] {re:mixin,re:classloading,pl:mixin:APP:supplementaries.mixins.json:AbstractBlockMixin,pl:mixin:APP:kubejs-common.mixins.json:BlockStateBaseMixin,pl:mixin:APP:ferritecore.blockstatecache.mixin.json:AbstractBlockStateMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerWorld.func_217441_a(ServerWorld.java:508) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerChunkProvider.func_241099_a_(ServerChunkProvider.java:367) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at java.util.ArrayList.forEach(ArrayList.java:1259) ~[?:1.8.0_292] {}
	at net.minecraft.world.server.ServerChunkProvider.func_217220_m(ServerChunkProvider.java:351) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_217207_a(ServerChunkProvider.java:326) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:333) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:851) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	... 5 more
[09:05:21] [Server thread/ERROR] [minecraft/MinecraftServer]: This crash report has been saved to: /Create- Above and Beyond 1.2/minecraft/crash-reports/crash-2021-12-10_09.05.21-server.txt
[09:05:21] [Server thread/INFO] [minecraft/MinecraftServer]: Stopping server
[09:05:21] [Server thread/INFO] [minecraft/MinecraftServer]: Saving players
[09:05:21] [Render thread/INFO] [STDOUT/]: [net.minecraft.util.registry.Bootstrap:func_179870_a:123]: ---- Minecraft Crash Report ----
// Oh - I know what I did wrong!

Time: 12/10/21 9:05 AM
Description: Exception ticking world

java.lang.ClassCastException: net.minecraft.world.gen.feature.DecoratedFeatureConfig cannot be cast to net.minecraft.world.gen.feature.BaseTreeFeatureConfig
	at net.minecraft.block.trees.Tree.func_230339_a_(SourceFile:29) ~[?:?] {re:classloading}
	at net.minecraft.block.SaplingBlock.func_226942_a_(SaplingBlock.java:43) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.block.SaplingBlock.func_225542_b_(SaplingBlock.java:33) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.block.AbstractBlock$AbstractBlockState.func_227034_b_(AbstractBlock.java:624) ~[?:?] {re:mixin,re:classloading,pl:mixin:APP:supplementaries.mixins.json:AbstractBlockMixin,pl:mixin:APP:kubejs-common.mixins.json:BlockStateBaseMixin,pl:mixin:APP:ferritecore.blockstatecache.mixin.json:AbstractBlockStateMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerWorld.func_217441_a(ServerWorld.java:508) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerChunkProvider.func_241099_a_(ServerChunkProvider.java:367) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at java.util.ArrayList.forEach(ArrayList.java:1259) ~[?:1.8.0_292] {}
	at net.minecraft.world.server.ServerChunkProvider.func_217220_m(ServerChunkProvider.java:351) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_217207_a(ServerChunkProvider.java:326) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:333) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:851) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:78) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_292] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.block.trees.Tree.func_230339_a_(SourceFile:29) ~[?:?] {re:classloading}
	at net.minecraft.block.SaplingBlock.func_226942_a_(SaplingBlock.java:43) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.block.SaplingBlock.func_225542_b_(SaplingBlock.java:33) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.block.AbstractBlock$AbstractBlockState.func_227034_b_(AbstractBlock.java:624) ~[?:?] {re:mixin,re:classloading,pl:mixin:APP:supplementaries.mixins.json:AbstractBlockMixin,pl:mixin:APP:kubejs-common.mixins.json:BlockStateBaseMixin,pl:mixin:APP:ferritecore.blockstatecache.mixin.json:AbstractBlockStateMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerWorld.func_217441_a(ServerWorld.java:508) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerChunkProvider.func_241099_a_(ServerChunkProvider.java:367) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at java.util.ArrayList.forEach(ArrayList.java:1259) ~[?:1.8.0_292] {}
	at net.minecraft.world.server.ServerChunkProvider.func_217220_m(ServerChunkProvider.java:351) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_217207_a(ServerChunkProvider.java:326) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:333) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:A}
-- Affected level --
Details:
	All players: 1 total; [ServerPlayerEntity['Bailongo'/293, l='ServerLevel[nuWorld 1.2]', x=-954.30, y=73.00, z=972.98]]
	Chunk stats: ServerChunkCache: 4326
	Level dimension: minecraft:overworld
	Level spawn location: World: (32,68,240), Chunk: (at 0,4,0 in 2,15; contains blocks 32,0,240 to 47,255,255), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 4412868 game time, 6586454 day time
	Level name: nuWorld 1.2
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: true
	Level weather: Rain time: 139828 (now: false), thunder time: 91023 (now: false)
	Known server brands: forge
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:851) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:78) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_292] {}
```

**Comment by SoftPoison** [comment]
Updating Forbidden and Arcanus to 16.2.2 seems to fix this issue https://github.com/stal111/Forbidden-Arcanus/issues/205

**Comment by astralstriker** [comment]
Same happens with Cherrywood Sapling from the same mod. Hoping for a hotfix from simibubi.

**Comment by oierbravo** [comment]
> Updating Forbidden and Arcanus to 16.2.2 seems to fix this issue [stal111/Forbidden-Arcanus#205](https://github.com/stal111/Forbidden-Arcanus/issues/205)

It works in my case.
Both bonemeal issue and natural grouth.

Thx!!!
